File: 2-4-1.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        Pair<Integer, Integer> answer = new Pair<>(0, 0);
        int n = scanner.nextInt();
        Deque<Integer> q = new ArrayDeque<>();
        for (int i = 0; i < n; i++) {
            int op = scanner.nextInt();
            if (op == 1) {
                int a = scanner.nextInt();
                q.addLast(a);
                if (answer.getFirst() < q.size() || (answer.getFirst() == q.size() && answer.getSecond() > q.getLast()))
                    answer = new Pair<>(q.size(), q.getLast());
            } else {
                q.removeFirst();
            }
        }
        System.out.println(answer.getFirst() + " " + answer.getSecond());
    }
}

class Pair<T, U> {
    private T first;
    private U second;

    public Pair(T first, U second) {
        this.first = first;
        this.second = second;
    }

    public T getFirst() {
        return first;
    }

    public U getSecond() {
        return second;
    }
}

File: 2-4-1.cpp
#include<bits/stdc++.h>
using namespace std;
typedef pair<int, int> pii;
int main() {
    pii answer = { 0, 0 };
    int n; cin >> n;
    deque<int> q;
    for (int i = 0; i < n; i++) {
        int op; cin >> op;
        if (op == 1) {
            int a; cin >> a;
            q.push_back(a);
            if (answer.first < q.size() || (answer.first == q.size() && answer.second > q.back()))
                answer = { q.size(), q.back() };
        }
        else {
            q.pop_front();
        }
    }
    cout << answer.first << ' ' << answer.second;
}